tika-parsers/src/main/java/org/apache/tika/parser/pkg/RarParser.java

```diff
@@ -86,7 +86,7 @@ public void parse(InputStream stream, ContentHandler handler,
                         subFile = rar.getInputStream(header);
 
                         Metadata entrydata = PackageParser.handleEntryMetadata(
-                                header.getFileNameString(),
+                                header.getFileNameW().equals("")?header.getFileNameString():header.getFileNameW(),

```

Comment:
Use "".equals(header.getFileName()) syntax, because this will cause NullPointerException if the wide file name is not available. Alternatively add a null check.